.github/workflows/publish.yaml

```diff
@@ -1,27 +1,27 @@
 name: Publish
 
 on:
-  push:
-    branches:
-      - master
-      - main
-      - track/**
-  pull_request:
-    branches:
-      - master
-      - main
-      - track/**
+  workflow_call:
+    secrets:
+      credentials:
+        required: true
 
 jobs:
+
   publish-charm:
     name: Publish Charm
     runs-on: ubuntu-latest
-    # Only publish to charmhub if we are pushing to a special branch or running PRs from something named `branch/*`
-    if: (github.event_name == 'push') ||  (startsWith( github.head_ref, 'branch/' ))
```

Comment:
Where is this logic now?

Reply:
I think we're abandoning this logic. We had a discussion on it in [template operator](https://github.com/canonical/template-operator/pull/24#discussion_r870248750). From now on, on `pull_request` we want to publish to `latest/edge/branchname` in charmhub no matter what was the github branch name.